src/mongocxx/private/database.hpp

```diff
@@ -34,6 +34,19 @@ class database::impl {
         name(std::move(name))
     {}
 
+    impl(const impl& i) : 
+        database_t{libmongoc::database_copy(i.database_t)},
+        client_impl{i.client_impl},
+        name{i.name} 
+    {}
+
+    impl& operator=(const impl& i) {
+        database_t = libmongoc::database_copy(i.database_t);
```

Comment:
actually - do we even use the copy assignment operator of the impl? it looks like we only ever use the copy constructor